tests/rule_based_profiler/test_rule_based_profiler.py

```diff
@@ -1358,3 +1364,35 @@ def test_add_rule_bad_rule(
     with pytest.raises(AttributeError) as e:
         profiler.add_rule(not_a_rule)
     assert "'dict' object has no attribute 'name'" in str(e.value)
+
+
+@mock.patch("great_expectations.data_context.data_context.DataContext")
+def test_run_with_expectation_suite_arg(mock_data_context: mock.MagicMock):
+    profiler: RuleBasedProfiler = RuleBasedProfiler(
+        name="my_rbp", data_context=mock_data_context, config_version=1.0
+    )
+    suite: ExpectationSuite = ExpectationSuite(
+        expectation_suite_name="my_expectation_suite"
+    )
+    result_suite: ExpectationSuite = profiler.run(expectation_suite=suite)
+
+    assert id(suite) == id(result_suite)
```

Comment:
shouldn't we test the functionality? That we provide a suite with at least one expectation and we get back a suite that contains that original expectation plus ones that were added via profiling?

Reply:
you make a very valid point - I do like this test for validating that we're mutating and existing object but will follow with a true test of the actual functionality.